tests/epyccel/test_arrays.py

```diff
@@ -3975,11 +3970,51 @@ def test_array_ndmin_2_order(language):
     check_array_equal(f1(d), f2(d))
     check_array_equal(f1(e), f2(e))
 
+##==============================================================================
+## TEST ITERATION
+##==============================================================================
+
+def test_iterate_slice(language):
+    f1 = arrays.iterate_slice
+    f2 = epyccel(f1, language = language)
+    i = randint(2, 10)
+    assert f1(i) == f2(i)
 
-#def teardown_module():
-#    import os, glob
-#    dirname  = os.path.dirname( arrays.__file__ )
-#    pattern  = os.path.join( dirname, '__epyccel__*' )
-#    filelist = glob.glob( pattern )
-#    for f in filelist:
-#        os.remove( f )
+def test_unpacking(language):
+    f1 = arrays.unpack_array
+    f2 = epyccel(f1, language = language)
+
+    arr = np.arange(3, dtype=int)
+    assert f1(arr) == f2(arr)
+
+    arr = np.arange(12, dtype=int).reshape((3,4))
+    x1, y1, z1 = f1(arr)
+    x2, y2, z2 = f1(arr)
+    check_array_equal(x1, x2)
+    check_array_equal(y1, y2)
+    check_array_equal(z1, z2)
+
+    arr = np.arange(24, dtype=int).reshape((3,4,2))
+    check_array_equal(x1, x2)
```

Comment:
Are we not missing two fundamental lines here?
```suggestion
    arr = np.arange(24, dtype=int).reshape((3,4,2))
    x1, y1, z1 = f1(arr)
    x2, y2, z2 = f2(arr)
    check_array_equal(x1, x2)
```

Reply:
:facepalm:

Reply:
[a120253](https://github.com/pyccel/pyccel/pull/1793/commits/a120253c4e27949b36c01be3d04e9dac9756f86b) + 7109367db